src/sentry/identity/oauth2.py

```diff
@@ -316,46 +284,59 @@ def get_token_params(self, code, redirect_uri):
             "client_secret": self.client_secret,
         }
 
+    def get_access_token(self, pipeline: IdentityPipelineT, code: str) -> Response:
+        data = self.get_token_params(code=code, redirect_uri=absolute_uri(_redirect_url(pipeline)))
+        verify_ssl = pipeline.config.get("verify_ssl", True)
+        return safe_urlopen(self.access_token_url, data=data, verify_ssl=verify_ssl)
+
     def exchange_token(
         self, request: HttpRequest, pipeline: IdentityPipelineT, code: str
     ) -> dict[str, str]:
         with record_event(
             IntegrationPipelineViewType.TOKEN_EXCHANGE, pipeline.provider.key
         ).capture() as lifecycle:
-            # TODO: this needs the auth yet
-            data = self.get_token_params(
-                code=code, redirect_uri=absolute_uri(_redirect_url(pipeline))
-            )
-            verify_ssl = pipeline.config.get("verify_ssl", True)
             try:
-                req = safe_urlopen(self.access_token_url, data=data, verify_ssl=verify_ssl)
-                body = safe_urlread(req)
-                content_type = req.headers.get("Content-Type", "").lower()
-                if content_type.startswith("application/x-www-form-urlencoded"):
```

Comment:
I don't see this handling anywhere, did this move? _Edit:_ Oh is this VSTS specific?

Reply:
this logic is now captured in `self.get_access_token` which is implemented in each `VSTSOAuth2CallbackView`